### DISCUSSION TITLE
CLI(typebox): Type Singular name

### DISCUSSION THREAD

**Discussion by fratzinger** [feathersjs/feathers#2969] [Ideas]
See the two typebox schemas generated by the cli. On the left, it's a 'custom' `messages` service that I created with `feathers g service`. On the right it's the default `users` service that got created by `feathers g app`:
<img width="1088" alt="image" src="https://user-images.githubusercontent.com/22286818/202295718-cf306c01-0325-4f3f-8916-28a020897a56.png">

See the difference between the namings. In almost every variable name it's plural (left) vs. singular (right):
- `messagesSchema` vs `userSchema`
- `$id: 'Messages'` (plural) vs. `$id: 'User'` (singular)
- `export type Messages` vs. `export type User`
- `messagesResolver`...

We should make it singular by default, right? Popular solutions:
- https://www.npmjs.com/package/pluralize
- https://www.npmjs.com/package/inflection

**Comment by daffl** [comment]
I've been trying to leave the naming to the end user. In this case if you pick `message` as the name and `messages` as the path then the naming would be the way you expect (see e.g. the generating service section [in the guide](https://dove.feathersjs.com/guides/basics/services.html#generating-a-service)).

![Generating messages](https://dove.feathersjs.com/assets/generate-service.58a876b0.png)

**Comment by fratzinger** [comment]
Ok, I oversaw that 🤦‍♂️

But in `services/index.ts` i would have expected the plural version:
```ts
export const services = (app: Application) => {
  app.configure(messages)
  app.configure(user)
  // All services will be registered here
}
```
At least that was my understanding of feathers v4:
- Anything that has to do with the service: plural
- Anything that has to do with the model/db/schema: singular

What do you think about that?

So in my head (and I remember it was the case in the old cli) the `serviceName` is like the `servicePath` with camelCase and stripped `/v1/sub/...`.

Don't get me wrong, this is such a tiny detail which anyone can easily change. But I think it's worth to think about the convention.

What do you think about adding another question? For example:
> What is the name of the model?

**Comment by fratzinger** [comment]
Next question: naming-convention for the services folder and containing files? It's `users` by default. I would have expect it to match the 'name of the service'.

**Reply by daffl** [reply]
Hm, it's the service path as well. I'm wondering if it's enough for now to document the naming conventions [in the CLI guide](https://feathersjs.com/guides/cli/service.html).
